CompuCell3D/core/CompuCell3D/plugins/FocalPointPlasticity/FocalPointPlasticityLinkInventoryBase.h

```diff
@@ -0,0 +1,233 @@
+				FocalPointPlasticityTrackerData fppd = _link->getFPPTrackerData(cell1);
+				itr = fppdSet.find(fppd);
+				if (itr != fppdSet.end()) fppdSet.erase(fppd);
+			}
+		}
+
+		FPPInventory_t* getCellLinkInventory(CellG* _cell) { 
+			return &cellLinkInventoryTrackerAccessor->get(_cell->extraAttribPtr)->linkInv;
+		}
+
+		CellG* getCellGFromConst(const CellG* _cell) {
+			if (!_cell) return (CellG*)(0);
+			else {
+				CellG* _cellNC = potts->getCellInventory().getCellByIds(_cell->id, _cell->clusterId);
+				return _cellNC;
+			}
+		}
+
+	public:
+		FPPLinkInventoryBase() {}
+		FPPLinkInventoryBase(BasicClassAccessor<FPPLinkInventoryTracker<LinkType> >* _cellLinkInventoryTrackerAccessor, Potts3D* _potts, BasicClassAccessor<FocalPointPlasticityTracker>* _focalPointPlasticityTrackerAccessor)
+		{
+			cellLinkInventoryTrackerAccessor = _cellLinkInventoryTrackerAccessor;
+			potts = _potts;
+			focalPointPlasticityTrackerAccessor = _focalPointPlasticityTrackerAccessor;
+		}
+		virtual ~FPPLinkInventoryBase() {}
+
+		BasicClassAccessor<FocalPointPlasticityTracker> * getFocalPointPlasticityTrackerAccessorPtr() { return focalPointPlasticityTrackerAccessor; }
+		BasicClassAccessor<FPPLinkInventoryTracker<LinkType> >* getFocalPointPlasticityCellLinkInventoryTrackerAccessorPtr() { return cellLinkInventoryTrackerAccessor; }
+
+		virtual FPPTrackerDataSet& getFPPTrackerDataSet(CellG* _cell) { return FPPTrackerDataSet(); }
+
+		virtual int getLinkInventorySize() { return linkInventory.size(); }
+		linkInventory_t& getContainer() { return linkInventory; }
+		virtual linkInventoryItr_t linkInventoryBegin() { return linkInventory.begin(); }
+		virtual linkInventoryItr_t linkInventoryEnd() { return linkInventory.end(); }
+		virtual void incrementIterator(linkInventoryItr_t& _itr) { ++_itr; }
+		virtual void decrementIterator(linkInventoryItr_t& _itr) { --_itr; }
+
+		// Add a link to the link inventory and update internals
+		void addToInventory(LinkType* _link) {
+			addTrackerData(_link);
+			addLinkNoChain(_link);
+			getCellLinkInventory(_link->getObj0())->addLinkNoChain(_link);
+			getCellLinkInventory(_link->getObj1())->addLinkNoChain(_link);
+		}
+		// Remove a link to the link inventory and update internals
+		void removeFromInventory(LinkType* _link) {
+			removeTrackerData(_link);
+			removeLinkNoChain(_link);
+			getCellLinkInventory(_link->getObj0())->removeLinkNoChain(_link);
+			getCellLinkInventory(_link->getObj1())->removeLinkNoChain(_link);
+			delete _link;
+		}
+
+		// Get link inventory list
+		FPPLinkList getLinkList() {
+			FPPLinkList fppLinkList;
+			for (linkInventoryItr_t itr = linkInventory.begin(); itr != linkInventory.end(); ++itr)
+				fppLinkList.push_back(itr->second);
+			return fppLinkList;
+		}
+		// Get link inventory list by cell
+		FPPLinkList getCellLinkList(CellG* _cell) { return getCellLinkInventory(_cell)->getLinkList(); }
+		FPPLinkList getCellLinkList(const CellG* _cell) {
```

Comment:
Do not oveload functions based on constantness

Reply:
Implemented in https://github.com/CompuCell3D/CompuCell3D/pull/361/commits/5e10487f5061e77cf90a34f83bd8d9f5231628a0